### PR TITLE
JDK-8282232: [Win] GetMousePositionWithPopup test fails due to wrong mouse position

### PR DESCRIPTION
**Pull request by honkar-jdk**
This test issue was due to race condition that occured when using `-Dsun.java2d.uiScale >= 2`.
As a result it led to incorrect mouse position being compared as opposed to the updated mouse position after the final mouse move event.

Following is the event log before and after fix.

**Before fix:**
```
1st mouse move
2nd mouse move
Frame-1 Mouse Event
Frame-2 Mouse Event
java.awt.Point[x=29,y=29]
```

**After fix:**
```
1st mouse move
2nd mouse move
Frame-1 Mouse Event
3rd mouse move
Frame-2 Mouse Event
java.awt.Point[x=50,y=50]
```

Earlier Frame-2's mouseMoved() was being triggered on Robot's 2nd mouse move instead of 3rd mouse move. To fix it, the mouseMotionListener for Frame-2 is now added after Robot's 2nd mouse move is processed to avoid the race condition.

The updated test fix is checked on multiple uiScales including fractional scales for windows platform.
CI testing works as expected on all platforms.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8282232](https://bugs.openjdk.org/browse/JDK-8282232): [Win] GetMousePositionWithPopup test fails due to wrong mouse position

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13380/head:pull/13380` \
`$ git checkout pull/13380`

Update a local copy of the PR: \
`$ git checkout pull/13380` \
`$ git pull https://git.openjdk.org/jdk.git pull/13380/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13380`

View PR using the GUI difftool: \
`$ git pr show -t 13380`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13380.diff">https://git.openjdk.org/jdk/pull/13380.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13380#issuecomment-1499770078)

### CODE REVIEW

File: test/jdk/java/awt/Mouse/GetMousePositionTest/GetMousePositionWithPopup.java [hunk 1]
```diff
@@ -90,24 +110,28 @@ public void mouseMoved(MouseEvent e) {
...
-                    public void mouseMoved(MouseEvent e)
-                    {
-                        Point positionInFrame2 = frame2.getMousePosition();
-                        int deltaX = Math.abs(50 - positionInFrame2.x);
-                        int deltaY = Math.abs(50 - positionInFrame2.y);
-                        if (deltaX > 2 || deltaY > 2) {
-                            throw new RuntimeException("Wrong position reported. Should be [50, 50] but was [" +
-                                    positionInFrame2.x + ", " + positionInFrame2.y + "]");
-                        }
+            }
```
Comment (prrace) [inline test/jdk/java/awt/Mouse/GetMousePositionTest/GetMousePositionWithPopup.java:113]: So are you confident the requested location will be honoured ?
I seem to remember a test framework fix in which you had to work quite hard to be sure of the real location.
And the implied "170-120=50" might be clearer if we had variables not just literal numbers ?

Comment (honkar-jdk) [reply test/jdk/java/awt/Mouse/GetMousePositionTest/GetMousePositionWithPopup.java:113]: @prrace I agree, adding PassFailJFrame's [syncLocationToWindowManager()](https://github.com/openjdk/jdk/blob/a8871f5d26e5cb42c031c7b736ec30b1b147a2bc/test/jdk/java/awt/regtesthelpers/PassFailJFrame.java#L343) to this test will make it more robust.
I'll change the literals to variables to make the context clear.

Comment (honkar-jdk) [reply test/jdk/java/awt/Mouse/GetMousePositionTest/GetMousePositionWithPopup.java:117]: Test updated.

**Comment by honkar-jdk** [comment]
@prrace Please review

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**